src/policy/ancestor_packages.h

```diff
@@ -0,0 +1,135 @@
+        bool skip{false};
+        /** Whether this transaction "dangles," i.e. we know nothing about it it because it is
+         * missing inputs or depends on another transaction that is missing inputs.
+         * This value can be set to true by calling SkipWithDescendants(). */
+        bool dangles{false};
+        /** This value starts as std::nullopt when we don't have any fee information yet. It can be
+         * updated by calling LinearizeWithFees() if this entry isn't being skipped. */
+        std::optional<uint32_t> mining_sequence;
+        /** (Modified) fees of this transaction. Starts as std::nullopt, can be updated using AddFeeAndVsize(). */
+        std::optional<CAmount> fee;
+        /** Virtual size of this transaction. Starts as std::nullopt, can be updated using AddFeeAndVsize(). */
+        std::optional<int64_t> vsize;
+
+        CTransactionRef tx;
+        /** Txids of all in-package ancestors. Populated in ctor and does not change.
+         * Use GetAncestorSet() to get ancestor sets with the skipped transactions removed.  */
+        std::set<uint256> ancestor_subset;
+        /** Txids of all in-package descendant. Populated in ctor and does not change. */
+        std::set<uint256> descendant_subset;
+        explicit PackageEntry(CTransactionRef tx_in) : tx(tx_in) {}
+
+        // Used to sort Txns(), FilteredTxns(), and result of GetAncestorSet(). Always guarantees
+        // topological sort to the best of our knowledge (see IsSorted()), and puts more
+        // incentive-compatible packages first if that information is available.
+        //
+        // If ancestor score-based linearization sequence exists for both transactions, the
+        // transaction with the lower sequence number comes first.
+        //    If there is a tie, the transaction with fewer in-package ancestors comes first (topological sort).
+        //       If there is still a tie, the transaction with the higher base feerate comes first.
+        // Otherwise, the transaction with fewer in-package ancestors comes first (topological sort).
+        bool operator<(const PackageEntry& rhs) const {
+            if (mining_sequence == std::nullopt || rhs.mining_sequence == std::nullopt) {
+                // If mining sequence is missing for either entry, default to topological order.
+                return ancestor_subset.size() < rhs.ancestor_subset.size();
+            } else {
+                if (mining_sequence.value() == rhs.mining_sequence.value()) {
+                    // Identical mining sequence means they would be included in the same ancestor
+                    // set. The one with fewer ancestors comes first.
+                    if (ancestor_subset.size() == rhs.ancestor_subset.size()) {
+                        // Individual feerate. This is not necessarily fee-optimal, but helps in some situations.
+                        // (a.fee * a.vsize > b.fee * a.vsize) is a shortcut for (a.fee / a.vsize > b.fee / b.vsize)
```

Comment:
In e82f4e7d7b41592b8891a2c6f9b961b2fc2ec51e "[txpackages] add AncestorPackage for linearizing packages"

typo

```suggestion
                        // (a.fee * b.vsize > b.fee * a.vsize) is a shortcut for (a.fee / a.vsize > b.fee / b.vsize)
```

Reply:
Fixed